src/gwt/src/org/rstudio/studio/client/common/r/roxygen/RoxygenHelper.java

```diff
@@ -0,0 +1,813 @@
+         JsArrayString other)
+   {
+      JsArrayInteger result = JsArray.createArray().cast();
+      for (int i = 0; i < self.length(); i++)
+         if (!contains(other, self.get(i)))
+            result.push(i);
+      return result;
+   }
+   
+   private static native final boolean contains(
+         JsArrayString array,
+         String object)
+   /*-{ 
+      for (var i = 0, n = array.length; i < n; i++)
+         if (array[i] === object)
+            return true;
+      return false;
+   }-*/;
+         
+   
+   private static native final JsArrayString extractRoxygenBlock(
+         AceEditorNative editor,
+         Range range)
+   /*-{
+      var lines = editor.getSession().getDocument().$lines;
+      return lines.slice(range.start.row, range.end.row);
+   }-*/;
+   
+   private JsArrayString listParametersInRoxygenBlock(
+         JsArrayString block,
+         Pattern pattern)
+   {
+      JsArrayString roxygenParams = JsArrayString.createArray().cast();
+      for (int i = 0; i < block.length(); i++)
+      {
+         String line = block.get(i);
+         Match match = pattern.match(line, 0);
+         if (match != null)
+            roxygenParams.push(match.getGroup(1));
+      }
+      
+      return roxygenParams;
+   }
+   
+   private Range getRoxygenBlockRange(int row)
+   {
+      while (row >= 0 && StringUtil.isWhitespace(editor_.getLine(row)))
+         row--;
+      
+      int blockEnd = row + 1;
+      
+      while (row >= 0 && RE_ROXYGEN.test(editor_.getLine(row)))
+         row--;
+      
+      if (row == 0 && !RE_ROXYGEN.test(editor_.getLine(row)))
+         row++;
+      
+      int blockStart = row + 1;
+      
+      return Range.fromPoints(
+            Position.create(blockStart, 0),
+            Position.create(blockEnd, 0));
+   }
+   
+   private boolean hasRoxygenBlock(Position position)
+   {
+      int row = position.getRow() - 1;
+      if (row < 0) return false;
+      
+      // Skip whitespace.
+      while (row >= 0 && StringUtil.isWhitespace(editor_.getLine(row)))
+         row--;
+      
+      // Check if we landed on an Roxygen block.
+      return RE_ROXYGEN.test(editor_.getLine(row));
+   }
+   
+   private void insertRoxygenTemplate(
+         String name,
+         JsArrayString argNames,
+         JsArrayString argTypes,
+         String argTagName,
+         String type,
+         Position position)
+   {
+      String typeCapitalized = StringUtil.capitalizeAllWords(type);
+      String roxygenParams = argsToExampleRoxygen(
+            argNames,
+            argTypes,
+            argTagName);
+      
+      // Add some spacing between params and the next tags,
+      // if there were one or more arguments.
+      if (argNames.length() != 0)
+         roxygenParams += "\n#'\n";
+      
+      String block = 
+                  "#' " + typeCapitalized + " '" + name + "'\n" +
+                  "#'\n" +
+                  "#' Provide a description of your " + type + " in the\n" +
```

Comment:
I think this could be shortened to just "Description" (and the next block to "Details" or whatever it actually translates to). My thought is that I don't think we want to make users do extra work (deleting lines) every time they insert a roxygen function just do we can deliver this help text once). I'll go ahead and merge this without these changes and you can commit them to master if you agree.